Dockerfile.assisted-test-infra-build

```diff
@@ -0,0 +1,18 @@
+# Generate python client
+FROM quay.io/edge-infrastructure/swagger-codegen-cli:2.4.18 AS swagger_py
+
+COPY assisted-service/swagger.yaml /
+COPY assisted-service/tools/generate_python_client.sh .
+RUN chmod +x ./generate_python_client.sh && SWAGGER_FILE=swagger.yaml OUTPUT=/build ./generate_python_client.sh
+
+FROM registry.access.redhat.com/ubi9/python-311:latest AS pybuilder
+
+COPY --chown=1001:0 ./assisted-service /assisted-service
+COPY --from=swagger_py --chown=1001:0 /build /assisted-service/build
+RUN python3 -m pip install -I pip && pip3 install -I -r /assisted-service/dev-requirements.txt
+RUN cd /assisted-service/build && python3 ../tools/client_package_initializer.py ./ https://github.com/openshift/assisted-service
```

Comment:
Currently we use upstream assisted-service-client by default. If the `SERVICE` env var is set, then it substitutes the old image with the new one. This way we get the correct client. my change achieves the same thing in a different way, using assisted-service code instead of substituting images